### ISSUE TITLE
Reader View fails on snorl.ax

### ISSUE THREAD

**Issue by SnorlaxYum** [mozilla/readability#550]
Tested with https://github.com/buriy/python-readability and found it able to parse posts like https://snorl.ax/browser/2019/05/10/refresh-the-browser/

However in Firefox it won't show the reader view button on the address bar of the page

**Comment by SnorlaxYum** [comment]
On the page https://snorl.ax/my-isso-style/config/ , not all the contents are loaded in the mode.

**Comment by gijsk** [comment]
> Tested with https://github.com/buriy/python-readability and found it able to parse posts like https://snorl.ax/browser/2019/05/10/refresh-the-browser/

This gives me a 403 forbidden error, as do all of the other snorl.ax links. Can you provide working links?

**Comment by SnorlaxYum** [comment]
> > Tested with https://github.com/buriy/python-readability and found it able to parse posts like https://snorl.ax/browser/2019/05/10/refresh-the-browser/
> 
> This gives me a 403 forbidden error, as do all of the other snorl.ax links. Can you provide working links?

Sorry that's due to my referer rule on html. Seems unnecessary. Just removed the rule.

**Comment by gijsk** [comment]
> Tested with https://github.com/buriy/python-readability and found it able to parse posts like https://snorl.ax/browser/2019/05/10/refresh-the-browser/
> 
> However in Firefox it won't show the reader view button on the address bar of the page. 

Yeah, the `isProbablyReaderable` code doesn't check the `<ul>` (which is most of the post) and thus doesn't think the contents are going to be readerable. That's why the button is not shown.

> Seems like Firefox only shows the button on the post with longer length like this https://snorl.ax/terminal/2019/06/13/root-flattening-cname-accuracy-simple-test/ . But the ul under the Table Of Contents won't show in the mode.

This is probably (haven't doublechecked) because the entire contents of the ToC are links. We ignore paragraphs with a link density that is too high, because it's likely a menu or other bit of content that's not part of the main article body. In this case, I think it's disputable whether that's a good outcome (ie, from this report, it seems you think it'd need to be included; I'm not convinced that's really the best here).

> On the page https://snorl.ax/my-isso-style/config/ , not all the contents are loaded in the mode.

It looks like here, we determine that a single list element has the highest score, and then end up ignoring everything else. Not 100% sure what the best fix is going to be there...
